.github/workflows/test-provider.yaml

```diff
@@ -32,8 +32,12 @@ on:
 
 jobs:
   test-render-providers:
-    # avoid running on PRs coming from a fork
-    if: github.event.pull_request.head.repo.full_name == github.repository || github.event_name != 'pull_request'
+    # Mechanism to prevent this test from running
+    # - on forks whose maintainers specify that provider credentials are absent (by setting a github variable NO_PROVIDER_CREDENTIALS)
+    # - on PRs coming from a fork (in which case github does not make the destination fork's credentials available)
```

Comment:
```suggestion
    # Prevents the execution of this test under the following conditions:
    # 1. When the 'NO_PROVIDER_CREDENTIALS' GitHub variable is set, indicating the absence of provider credentials.
    # 2. For pull requests (PRs) originating from a fork, since GitHub does not provide the fork's credentials to the destination repository.
```